### ISSUE TITLE
Not able to log the messages inside the "OnStop" event in windowService

### ISSUE THREAD

**Issue by BhanuPG** [NLog/NLog#2031]
**Type** (choose one):- Bug
**NLog version**: (e.g. 4.4.3)

**Platform**:  .Net 4.0 

Hi previously I was using NLog 2.0.0 to log the messages and I was able to log the messages from both OnStart ,OnStop events in WindowsService. Below is the code inside OnStart, OnStop events

 protected override void OnStart(string[] args)
        { 
            ImintLogger.Post(Severity.Warning,  "My service started");
            _stop = false;    
        }

 protected override void OnStop()
        {    
            Logger.Post(Severity.Warning,"My service Stopped");
            DisableTimer();
            _stop = true;
            _timer.Dispose();
            _timerBackendOps.Dispose();
            _timerHealthCheck.Dispose();
            _timerHeartBeatCheck.Dispose();            
        }
 But after upgrading NLog to 4.4.3 I'm not able to log the messages from OnStop event.
I didn't add NLog config as it's working fine for remaining modules.

**Comment by UgurAldanmaz** [comment]
Hi @BhanuPG. Could you add your configration here?

**Comment by BhanuPG** [comment]
Hi @UgurAldanmaz  below is the configuration.
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <targets async="true">
    
    <target name="file" xsi:type="File" fileName="C:/Temp/log/service_${shortdate}.txt" archiveFileName="C:/Temp/log/service_${shortdate}.txt" archiveAboveSize="50000000" archiveEvery="Day" archiveNumbering="Rolling" maxArchiveFiles="30" concurrentWrites="true" keepFileOpen="false" encoding="iso-8859-2" layout="${longdate}|${level:uppercase=true}|${logger}|${event-context:item=RssGroup}|${message}}|${exception:format=tostring:maxInnerExceptionLevel=3}" />
    <target name="database" xsi:type="Database" connectionStringName="IMINTConnection">
      <commandText>
        EXEC [dbo].[insertLogMessagesToDB] @rssGroup, @moduleName, @severity, @detail, @isSummary
      </commandText>
      <parameter name="@rssGroup" layout="${event-context:item=RssGroup}" />
      <parameter name="@moduleName" layout="${logger}" />
      <parameter name="@severity" layout="${level:uppercase=true}" />
      <parameter name="@detail" layout="${message}" />
      <parameter name="@isSummary" layout="${event-context:item=IsSummary}" />
    </target>
  </targets>
  <rules>
    <logger name="imintlog" minlevel="Debug" writeTo="file" />
    <logger name="imintlog.db" minlevel="Info" writeTo="database" />
  </rules>
</nlog>

**Comment by UgurAldanmaz** [comment]
Also internal log would be helpful. If you don't know how can you find it, [wiki page](https://github.com/NLog/NLog/wiki/Internal-logging) can help you.

**Comment by snakefoot** [comment]
I'm guessing duplicate of #1998

**Comment by 304NotModified** [comment]
thanks @UgurAldanmaz and @snakefoot !

@snakefoot thinking to just publish 4.4.5 without waiting on confirmation. If it's isn't OK, will could always release a 4.4.6. Agree?

**Comment by 304NotModified** [comment]
I have released 4.4.5. https://www.nuget.org/packages/NLog/4.4.5 

I'm pretty sure this is fixing your issue. Please let us know!
If this isn't fixing your issue, we will reopen the issues.

**Comment by BhanuPG** [comment]
Thank you very much... My problem has resolved. Now I'm able to log the messages.

**Comment by 304NotModified** [comment]
Thanks for the confirm!
